### ISSUE TITLE
Relax root overlap requirements for `buf protoc`

### ISSUE THREAD

**Issue by bufdev** [bufbuild/buf#351]
See https://github.com/cockroachdb/cockroach/pull/66913#issuecomment-870092892

```
$ protoc -o /dev/null -I proto -I . $(buf ls-files proto)
buf/alpha/registry/v1alpha1/resolve.proto:20:1: warning: Import buf/alpha/registry/v1alpha1/module.proto is unused.
$ buf protoc -o /dev/null -I proto -I . $(buf ls-files proto)
include directory "/home/foo/dev/buf/proto" is within include directory "/home/foo/dev/buf" which is not allowed
```

**Comment by bufdev** [comment]
Notes to self:

```
# takes the first include that it is in
$ protoc -o /dev/stdout -I . -I pkg  pkg/a/a.proto | buf build -o -#format=json -
{"file":[{"name":"pkg/a/a.proto","package":"a","messageType":[{"name":"Foo"}],"syntax":"proto3"}],"bufbuildImageExtension":{}}
$ buf protoc -o /dev/stdout -I . -I pkg  pkg/a/a.proto | buf build -o -#format=json -
{"file":[{"name":"pkg/a/a.proto","package":"a","messageType":[{"name":"Foo"}],"syntax":"proto3"}],"bufbuildImageExtension":{}}

# takes the first include that it is in
$ protoc -o /dev/stdout -I pkg -I .  pkg/a/a.proto | buf build -o -#format=json -
{"file":[{"name":"a/a.proto","package":"a","messageType":[{"name":"Foo"}],"syntax":"proto3"}],"bufbuildImageExtension":{}}
# does *not* take the first include that it is in
$ buf protoc -o /dev/stdout -I pkg -I .  pkg/a/a.proto | buf build -o -#format=json -
{"file":[{"name":"pkg/a/a.proto","package":"a","messageType":[{"name":"Foo"}],"syntax":"proto3"}],"bufbuildImageExtension":{}}

# takes the only include that it is in
$ protoc -o /dev/stdout -I pkg -I .  a/a.proto | buf build -o -#format=json -
{"file":[{"name":"a/a.proto","package":"a","messageType":[{"name":"Foo"}],"syntax":"proto3"}],"bufbuildImageExtension":{}}
# takes the only include that it is in
$ buf protoc -o /dev/stdout -I pkg -I .  a/a.proto | buf build -o -#format=json -
{"file":[{"name":"a/a.proto","package":"a","messageType":[{"name":"Foo"}],"syntax":"proto3"}],"bufbuildImageExtension":{}}

```

**Comment by bufdev** [comment]
#353 is strongly related to this and will likely be fixed simultaneously.

**Comment by bufdev** [comment]
We have now deleted `buf protoc` by moving it to alpha in #915, and will likely eventually delete it entirely. As such, going to close this issue as `buf protoc` will almost for sure not be worked on in the future. If we decide to revisit this decision, we should reopen this issue. All issues that were closed are tracked on the #915.
